Framework/src/InfrastructureGenerator.cxx

```diff
@@ -467,9 +468,13 @@ void InfrastructureGenerator::generateCheckRunners(framework::WorkflowSpec& work
     ILOG(Info) << DataSpecUtils::describe(output) << " ";
   ILOG(Info) << ENDM;
 
-  // as we need information about the checkRunners we do it here instead of a separate static method
-  AggregatorRunnerFactory aggregatorRunnerFactory;
-  DataProcessorSpec spec = aggregatorRunnerFactory.create(checkRunnerOutputs, configurationSource);
+  return checkRunnerOutputs;
+}
+
+void InfrastructureGenerator::generateAggregator(WorkflowSpec& workflow, std::string configurationSource, vector<framework::OutputSpec>& checkRunnerOutputs)
+{
+  // TODO consider whether we should recompute checkRunnerOutputs instead of receiving it all baked.
+  DataProcessorSpec spec = AggregatorRunnerFactory::create(checkRunnerOutputs, configurationSource);
```

Comment:
As an improvement, we probably should use only outputs which are needed by the aggregator

Reply:
absolutely: https://alice.its.cern.ch/jira/browse/QC-492